go-controller/pkg/node/node.go

```diff
@@ -595,6 +604,103 @@ func (n *OvnNode) WatchEndpoints() error {
 	return err
 }
 
+func exGatewayPodsAnnotationsChanged(oldNs, newNs *kapi.Namespace) bool {
+	// In reality we only care about exgw pod deletions, however since the list of IPs is not expected to change
+	// that often, let's check for *any* changes to these annotations compared to their previous state and trigger
+	// the logic for checking if we need to delete any conntrack entries
+	return (oldNs.Annotations[util.ExternalGatewayPodIPsAnnotation] != newNs.Annotations[util.ExternalGatewayPodIPsAnnotation]) ||
+		(oldNs.Annotations[util.RoutingExternalGWsAnnotation] != newNs.Annotations[util.RoutingExternalGWsAnnotation])
+}
+
+func (n *OvnNode) checkAndDeleteStaleConntrackEntries() {
+	namespaces, err := n.watchFactory.GetNamespaces()
+	if err != nil {
+		klog.Errorf("Unable to get pods from informer: %v", err)
+	}
+	for _, namespace := range namespaces {
+		if (len(namespace.Annotations[util.ExternalGatewayPodIPsAnnotation]) > 0) || (len(namespace.Annotations[util.RoutingExternalGWsAnnotation]) > 0) {
```

Comment:
@tssurya hmm; do we need this code to trigger when all gateways are removed from the namespace? Or is that handled somewhere else?

Reply:
when all gateways are removed from namespace we would hit the oldNs != newNs below for the event triggered mechanism right? after that we don't need any periodic ARPs for empty annotations right?

Reply:
however I could just change this to "if namespace is annotated" if that makes it better

Reply:
actually you have a point. I don't think we ever do a allowMACList={empty} pass to deletion filter. :/